pkg/controllerhelpers/core_test.go

```diff
@@ -358,3 +423,105 @@ func TestAggregateStatusConditions(t *testing.T) {
 		})
 	}
 }
+
+func TestJoinWithLimit(t *testing.T) {
+	const (
+		testSep                  = ","
+		testSepLen               = 1
+		testLimit                = 100
+		testOmissionIndicatorFmt = " (%d element(s) omitted)"
+	)
+
+	tt := []struct {
+		name                 string
+		elems                []string
+		expected             string
+		omissionIndicatorFmt string
+	}{
+		{
+			name:     "empty slice",
+			elems:    []string{},
+			expected: "",
+		},
+		{
+			name:     "nil slice",
+			elems:    nil,
+			expected: "",
+		},
+		{
+			name: "single element exceeds limit",
+			elems: []string{
+				strings.Repeat("a", testLimit+1),
+			},
+			expected: "(1 element(s) omitted)",
+		},
+		{
+			name: "omission indicator doesn't include %d placeholder",
+			elems: []string{
+				strings.Repeat("a", testLimit+1),
+			},
+			omissionIndicatorFmt: "(some elements omitted)",
+			// Accepting this as an expected result - as long as it doesn't panic, it's fine.
```

Comment:
Not sure about this, seems very bug prone given the only safeguard is the comment explaining the expected format. It's only an internal func so it's OKish, but could the function verify the format string somehow?

Reply:
I see your point. As you mentioned, given this is an internal function, I decided to treat this as an accepted result as long as it doesn't panic. I'm happy to make this more robust by sacrificing the custom fmt and always using `(%d more omitted)` to be 100% sure we don't do anything harmful in the future.